src/neo-vm/ExecutionEngine.cs

```diff
@@ -346,15 +345,19 @@ private bool ExecuteInstruction()
                         }
                     case OpCode.XDROP:
                         {
-                            int n = (int)context.EvaluationStack.Pop().GetBigInteger();
+                            if (!(context.EvaluationStack.Pop() is PrimitiveType item_n))
```

Comment:
`bool TryPop<T>()` ?

` if (!(context.EvaluationStack.TryPop<PrimitiveType>(out var item_n))`

Reply:
We can avoid some exceptions with https://github.com/neo-project/neo-vm/pull/224

Reply:
Why we need to avoid the exceptions?

Reply:
Because is a waste of time, you will return false in both cases, the difference is that throw exceptions have a big performance penalty.

Reply:
It does get faster. But since there has been an exception, the contract will stop running, so there is no difference in the speed of a single. On the contrary, it is more convenient to use exceptions.

Reply:
I am not agree @erikzhang , throw exceptions have a big computation cost. We should try to avoid it everything is possible

This is the case , it will return `false`, but capturing the exception

```csharp
bool WithoutException() { return false; }
bool WithException() { try { throw new Exception(); } catch { return false; } }

        [TestMethod]
        public void Test()
        {
            var st1 = Stopwatch.StartNew();

            for (int x = 0; x < 10_000; x++)
            {
                try { WithException(); } catch { }
            }

            st1.Stop();
            var st2 = Stopwatch.StartNew();

            for (int x = 0; x < 10_000; x++)
            {
                try { WithoutException(); } catch { }
            }

            st2.Stop();

            Console.WriteLine("With exception: " + st1.Elapsed.ToString());
            Console.WriteLine("Without exception: " + st2.Elapsed.ToString());
        }
```


```
With exception: 00:00:00.1952785
Without exception: 00:00:00.0000160
```

Reply:
If you run it in release mode, it will be much faster.

Reply:
We have many `try` and `catch`. If you want, we can optimize them in another PR.

Reply:
I think that this was in Release, but is ok for me do that in a different PR